frontend/actions/src/main/java/com/virtuslab/gitmachete/frontend/actions/base/BaseCheckoutAction.java

```diff
@@ -88,23 +80,9 @@ public void actionPerformed(AnActionEvent anActionEvent) {
     val gitRepository = getSelectedGitRepository(anActionEvent);
 
     if (gitRepository != null) {
-
       log().debug(() -> "Queuing '${targetBranchName}' branch checkout background task");
-      new Task.Backgroundable(project, getString("action.GitMachete.BaseCheckoutAction.task-title")) {
-        @Override
-        @UIThreadUnsafe
-        public void run(ProgressIndicator indicator) {
-          doCheckout(project, indicator, targetBranchName, gitRepository);
-        }
-      }.queue();
+      GitBrancher.getInstance(project).checkout(/* reference */ targetBranchName, /* detach */ false,
```

Comment:
As per ArchUnit test failure... TBD if `GitBrancher` methods can be safely called from UI thread

Reply:
:yes: because the GitBrancher methods wrap actions within `git4idea.branch.GitBrancherImpl.CommonBackgroundTask` which is a usual `Task.Backgroundable`, and `runInBackground` simply does `task.queue()`